editor/index.js

```diff
@@ -25,6 +26,19 @@ export function createEditorInstance( id, post ) {
 		blocks: wp.blocks.parse( post.content.raw ),
 	} );
 
+	// Each property that is set in `post-content.js` (other than `content`
+	// because it is serialized when a save is requested) needs to be
+	// registered as an edit now.  Otherwise the initial values of these
+	// properties will not be properly saved with the post.
+	store.dispatch( {
+		type: 'EDIT_POST',
+		edits: {
+			title: post.title.raw,
+			...omit( post, 'title', 'content' ),
+		},
+		shouldMarkDirty: false,
+	} );
```

Comment:
Trying to think broadly. I see `edits` as an object of attributes explicitly modified by the user, IMO I don't think we should do this because it creates confusion on what the `edit` reducer contains.

That said, I've already noticed the bug and as you said, the need for this may disappear when we'll drop the `post-content` file. An alternative could be to merge the `edits` with `currentPost` in the `savePost` action instead.